firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Protocols/Messages.tpp

```diff
@@ -19,8 +19,11 @@ template <typename TaggedUnion, typename MessageTypes, size_t max_size>
 template <size_t output_size, size_t num_descriptors>
 MessageStatus Message<TaggedUnion, MessageTypes, max_size>::write(
     Util::ByteVector<output_size> &output_buffer,
-    const Util::ProtobufDescriptors<num_descriptors> &pb_protobuf_descriptors) const {
-  auto type = static_cast<uint8_t>(payload.tag);
+    const Util::ProtobufDescriptors<num_descriptors> &pb_protobuf_descriptors) {
+  static_assert(
+      Util::ByteVector<output_size>::max_size() >= max_size,
+      "Write method unavailable as output buffer is too small");
+  type = static_cast<uint8_t>(payload.tag);
   if (type > pb_protobuf_descriptors.size()) {
```

Comment:
Should this be `>` or `>=`? In `parse` we changed `>` to `>=` because `>` was an off-by-one error. In either case, let's add a unit test for this edge case, for both `parse` and `write`.

Reply:
> Should this be > or >=? In parse we changed > to >= because > was an off-by-one error.

Yes, I think so too, updated. added for both with `invalid_type `result.